templates/accounts/account.html

```diff
@@ -322,37 +322,46 @@
             google.maps.event.addListener(infowindow, 'closeclick', function() {
                 stopAll();
             });
-
-            ajaxLoad('{% url "geotags-for-user-latest-json" user.username %}', function(data, responseCode)
-            {
+            
+            var oReq = new XMLHttpRequest();
+            oReq.open("GET", '{% url "geotags-for-user-latest-barray" user.username %}', true);
+            oReq.responseType = "arraybuffer";
+            oReq.onload = function(oEvent) {
                 var bounds = new google.maps.LatLngBounds();
                 var nSounds = 0;
                 var lastPoint;
-
-                $.each($.parseJSON(data.response), function(index, item)
-                {
-                    nSounds = nSounds + 1;
-                    var id = item[0];
-                    var lat = item[1];
-                    var lon = item[2];
-
-                    var point = new google.maps.LatLng(lat, lon);
-                    lastPoint = point;
-                    bounds.extend(point);
-                    var marker = new google.maps.Marker({'position': point, 'map': map});
-
-                    google.maps.event.addListener(marker, 'click', function()
-                    {
-                        stopAll();
-                        ajaxLoad( '/geotags/infowindow/' + id, function(data, responseCode)
+  
+                var raw_data = new Int32Array(oReq.response);
+
+                var id = null;
+                var lat = null;
+                var lon = null;
+
+                $.each(raw_data, function(index, item) {
```

Comment:
and shouldn't this functionality be put into a separate class/function? It will be repeated all overt the place and the "1000" magic number is going to appear everywhere...